certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -193,6 +198,8 @@ def choose_vhost(self, target_name):
         :rtype: :class:`~certbot_nginx.obj.VirtualHost`
 
         """
+        if self.vhost is not None:
```

Comment:
Anyway, I realize that if we're doing this it should be cached by nameset or something. It's probably best to not do this particular thing for now, and maybe add some other type of caching as a feature later because it will require more careful thought.